Makefile

```diff
@@ -0,0 +1,30 @@
+SOURCE_FILES?=$$(glide novendor)
+TEST_PATTERN?=.
+TEST_OPTIONS?=
+
+setup: ## Install all the build and lint dependencies
+	curl https://glide.sh/get | sh
+	go get -u github.com/kisielk/errcheck
+	go get -u github.com/golang/lint/golint
+	glide install
+
+test: ## Run all the tests
+	go test $(TEST_OPTIONS) -cover $(SOURCE_FILES) -run $(TEST_PATTERN) -timeout=30s
+
+lint: ## Run all the linters
+	errcheck $(SOURCE_FILES)
+	go vet $(SOURCE_FILES)
+	golint ./... | grep -v vendor
+
+ci: lint test ## Run all the tests and code checks
+
+build: ## Build a beta version of releaser
+	go build
```

Comment:
> One good example is adding custom building IDs to binaries.

Thats a fair point, though arguably thats what this project does anyway. Is there value to building multiple versions on a local machine to check compilation?

A live, released version is either going to be installed via `go get -u ...` for HEAD, or via https://github.com/goreleaser/get. Realistically this step will only be used, barring edges, to test the tool compiles and to blackbox it before push/PR/merge.

Having the goal here wont make much difference to people using `go build`, of course; I'm just wondering whether using the golang toolchain and something like github.com/chuyeow/golang-pre-commit-hooks to check linters and stuff on commit would fit the workflow better.